.gitlab-ci.yml

```diff
@@ -1,197 +1,152 @@
 stages:
   - test
-  - push-release
   - build
 variables:
   RUST_BACKTRACE: "1"
   RUSTFLAGS: ""
   CARGOFLAGS: ""
   CI_SERVER_NAME: "GitLab CI"
   LIBSSL: "libssl1.0.0 (>=1.0.0)"
-  CARGO_HOME: $CI_PROJECT_DIR/cargo
 cache:
-  key: "$CI_BUILD_STAGE-$CI_BUILD_REF_NAME"
+  key: "$CI_JOB_NAME"
   paths:
     - target/
     - cargo/
   untracked: true
-linux-stable:
+.list_of_refs: &list_of_refs #list of git refs for build and/or push, deploy
   stage: build
-  image: parity/rust:gitlab-ci
   only:
+    - stable
     - beta
+    - gitlab
     - tags
-    - stable
     - triggers
+.artifacts_rules: &artifacts_rules
+  artifacts:
+    when: on_success
+    expire_in: 1 mos
```

Comment:
Actually might be better to just keep fixed number of artifacts (i.e. might be good to download old stable binaries even after a month).

Reply:
we have them in S3 and vanity